R/check.R

```diff
@@ -34,17 +34,27 @@ check_built_site <- function(pkg = ".") {
 
 check_missing_images <- function(pkg, src_path, dst_path) {
   html <- xml2::read_html(path(pkg$dst_path, dst_path), encoding = "UTF-8")
-  src <- xml2::xml_attr(xml2::xml_find_all(html, ".//img"), "src")
+  img <- xml2::xml_find_all(html, ".//img")
 
+  src <- xml2::xml_attr(img, "src")
   rel_src <- src[xml2::url_parse(src)$scheme == ""]
   rel_path <- fs::path_norm(path(fs::path_dir(dst_path), rel_src))
   exists <- fs::file_exists(path(pkg$dst_path, rel_path))
-
   if (any(!exists)) {
     paths <- rel_src[!exists]
-    cli::cli_warn(c(
-      "Missing images in {.file {path_rel(src_path, pkg$src_path)}}: {.file {paths}}",
+    cli::cli_inform(c(
+      x = "Missing images in {.file {path_rel(src_path, pkg$src_path)}}: {.file {paths}}",
       i = "pkgdown can only use images in {.file man/figures} and {.file vignettes}"
     ))
   }
+
+  alt <- xml2::xml_attr(img, "alt")
+  if (any(is.na(alt))) {
+    problems <- src[is.na(alt)]
+    problems[grepl("^data:image", problems)] <- "<base64 encoded image"
+    cli::cli_inform(c(
```

Comment:
Do you see any downside to turning this to a warning instead?

The reason a warning might be preferable is that, in the CI context, a workflow can be set to turn warnings into errors. This can help detect either already existing or newly added images in a PR that don't have the alt-text, motivating some action. Otherwise, these messages can be quite easy to miss, esp. if they are appearing in the midst of other verbose processes, and can just continue to accumulate.

Reply:
I don't see warnings appearing much later as a feature as it generally makes it hard to track down to their source. I think it's generally fine for these issues to accumulate until you make a consistent effort to track them down (especially since this is something new that we're adding).